core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -111,7 +116,12 @@ public AtomicFileWriter(@Nonnull Path destinationPath, @Nonnull Charset charset)
         }
 
         try {
-            tmpPath = Files.createTempFile(dir, "atomic", "tmp");
+            Set<PosixFilePermission> permissions = Jenkins.DEFAULT_FILE_PERMISSIONS;
+            if (Files.exists(destinationPath)) {
+                permissions = Files.getPosixFilePermissions(destinationPath);
+            }
+            tmpPath = Files.createTempFile(dir, "atomic", "tmp", PosixFilePermissions.asFileAttribute(permissions));
```

Comment:
javadoc does not list `PosixFilePermissions.asFileAttribute` as null safe not document what happens if null is passed in.